### PR TITLE
Report errors to Sentry, not just print in logs

### PR DESCRIPTION
**Pull request by aakritigupta**
Fixes: https://github.com/travis-pro/team-teal/issues/1796

### CODE REVIEW

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
This is super awesome! Would it be possible to let some of these bubble up, by removing the begin/rescue?

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
I think in some cases the existing code catches the errors for a reason, so simply raising exceptions may not be safe.

**Comment by aakritigupta** [comment]
I tried to leave the rest of the logic intact, since I haven't really worked on hub. I'd prefer to leave the errors for rescue, since someone designed this way

**Comment by aakritigupta** [comment]
> I think in some cases the existing code catches the errors for a reason, so simply raising exceptions may not be safe.

I left some of them out on this purpose. But out of the `puts` replacements I have done, which one do you think shouldn't be raised?

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
I believe that re-raising the exception will change the behaviour in most of these cases. e.g. reroute will fail completely instead of returning `-1`, amqp errors will stop execution instead of gracefully falling through.

It would be safer to send the exception to sentry directly via `Raven. capture_exception(e)`. Based on the [example I found in the docs](https://github.com/getsentry/raven-ruby#call).

**Comment by aakritigupta** [comment]
Makes sense @igorwwwwwwwwwwwwwwwwwwww 
I edited, to log the strings as was done before, only with `logger` now and capturing exceptions in Raven, so that the logic remains the same, except for additionally, reporting to Sentry.

File: lib/travis/hub/amqp/queue.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@ module Travis
   module Hub
     class Amqp
       class Queue
+        class AmqpQueueError < StandardError; end
```
Comment (igorwwwwwwwwwwwwwwwwwwww) [inline lib/travis/hub/amqp/queue.rb]: this can be cleaned up

**Pull request merged**